docs/site/Where-filter.md

```diff
@@ -0,0 +1,611 @@
+---
+title: 'Where filter'
+lang: en
+keywords: LoopBack 4.0, LoopBack 4, where
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Where-filter.html
+summary: A <i>where</i> filter specifies a set of logical conditions to match, similar to a WHERE clause in a SQL query.
+---
+
+## REST API
+
+In the first form below, the condition is equivalence, that is, it tests whether _property_ equals _value_.
+The second form below is for all other conditions.
+
+```
+filter[where][property]=value
+```
+
+```
+filter[where][property][op]=value
+```
+
+Where:
+
+- _property_ is the name of a property (field) in the model being queried.
+- _value_ is a literal value. 
+- _op_ is one of the [operators](#operators) listed below.
+
+For example, if there is a `Car` model with an `odo` property, the following query finds instances where the `odo` is greater than 5000:
+
+```
+/cars?filter[where][odo][gt]=5000
+```
+
+For example, here is a query to find cars with `odo` is less than 30,000:
+
+```
+/cars?filter[where][odo][lt]=30000
+```
+
+You can also use [stringified JSON format](Querying-data.html#using-stringified-json-in-rest-queries) in a REST query.
+
+### Filter limit
+
+{% include important.html content="There is a limit of twenty filters (combined with AND or OR) using this format, due to the use of [qs](https://github.com/ljharb/qs#parsing-arrays).  When there are more than twenty, the filter is converted into an `Object` where it is expecting
+an `Array`. See [LoopBack issue #2824](https://github.com/strongloop/loopback/issues/2824) for more details.
+" %}
+
+You can encode the large filter object as "stringified JSON":
+
+```
+http://localhost:3000/api/Books
+?filter={"where":{"or":[{"id":1},{"id":2},...,{"id":20"},{"id":21}]}}
+```
+
+<!-- DOES NOT WORK IN LB4
```

Comment:
Do we need this commented code?

Reply:
From what I tested, this doesn't work in LB4, but we might want similar features in LB4. I use the comment as a reminder.